core-rust/state-manager/src/types.rs

```diff
@@ -395,7 +395,11 @@ impl UserPayloadHash {
     pub const LENGTH: usize = 32;
 
     pub fn for_transaction(transaction: &NotarizedTransaction) -> Self {
-        Self(blake2b_256_hash(manifest_encode(transaction).unwrap()).0)
+        Self::for_manifest_encoded_transaction(manifest_encode(transaction).unwrap())
+    }
+
+    pub fn for_manifest_encoded_transaction<T: AsRef<[u8]>>(transaction: T) -> Self {
```

Comment:
Strictly speaking this is a `manifest_encoded_notarized_transaction` (rather than a `manifest_encoded_ledger_transaction`) - we've had a few issues with confusing these - but maybe I'm being overly stickly